apps/studio/components/interfaces/DataWarehouse/CreateWarehouseAccessToken.tsx

```diff
@@ -0,0 +1,69 @@
+import { set } from 'lodash'
+import React from 'react'
+import { Button, Input, Modal } from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+
+type Props = {
+  onSubmit: (values: { description: string }) => Promise<void>
```

Comment:
any reason why we're passing onSubmit as a prop, and not just calling the mutation in this component directly?